engine/src/main/java/org/terasology/logic/delay/DelayedActionSystem.java

```diff
@@ -104,15 +107,18 @@ private void invokePeriodicOperations(long currentWorldTime) {
         operationsToInvoke.stream().filter(EntityRef::exists).forEach(periodicEntity -> {
             final PeriodicActionComponent periodicActionComponent = periodicEntity.getComponent(PeriodicActionComponent.class);
 
-            final Set<String> actionIds = periodicActionComponent.getTriggeredActionsAndReschedule(currentWorldTime);
-            saveOrRemoveComponent(periodicEntity, periodicActionComponent);
+            // If there is a PeriodicActionComponent, proceed.
```

Comment:
I would go for a guard statement instead of indenting everything by 1. (e.g. 
if (isError) {
    return;
}
normalHandling();

I am also not familiar with this code, but I think there should at least be a comment why null is a common case and not an error. If it is an error that is worth investigating I suggest you change the code to log an error.